index.js

```diff
@@ -6,13 +6,19 @@ var through = require("through2");
 // # Filter
 // Create a through stream that only passes data that passes the given test
 // function
-module.exports = function(test, opt) {
+var filter = module.exports = function(test, opt) {
 	return through(opt || {}, function(data, _, cb) {
 		if (test(data)) cb(null, data);
 		else cb(null);
 	});
 };
 
+module.exports.obj = function(test, opt) {
+	opt = opt || {}
+	opt.objectMode = true
```

Comment:
> use `xtend` to not mutate the passed argument […].

Where do I change the passed argument?

> […] set `highWaterMark`.

Just curious: Why a non-default value?